src/windows.rs

```diff
@@ -0,0 +1,35 @@
+use std::ptr::addr_of_mut;
+
+use winapi::shared::guiddef::GUID;
+
+use crate::ffi;
+use crate::{HidDevice, HidResult};
+
+impl HidDevice {
+    /// **`Only available on Windows`**
+    ///
+    /// Get the container ID for a HID device.
+    ///
+    /// This function returns the `DEVPKEY_Device_ContainerId` property of the
+    /// given device. This can be used to correlate different interfaces/ports
+    /// on the same hardware device.
+    pub fn get_container_id(&self) -> HidResult<GUID> {
+        unsafe {
+            let mut container_id: GUID = std::mem::zeroed();
+
+            let res = ffi::windows::hid_winapi_get_container_id(
+                self._hid_device,
+                addr_of_mut!(container_id),
+            );
+
+            if res == -1 {
```

Comment:
Does not have to be in unsafe block, prefer smaller isolated unsafe blocks. (even if there have to be two)

Reply:
Done! You can see the [changes here](https://github.com/ruabmbua/hidapi-rs/compare/02a62044aaa72eeea404f30415fc6f7fd3d50e18..4dac1a20d7b4be3df307040a980a4abdbb21a87b)